programs/rodeos/streams/rabbitmq.hpp

```diff
@@ -20,24 +20,24 @@ class rabbitmq : public stream_handler {
    std::vector<eosio::name>             routes_;
 
  public:
-   rabbitmq(boost::asio::io_service& io_service, std::vector<eosio::name> routes, std::string address, std::string queue_name)
+   rabbitmq(boost::asio::io_service& io_service, std::vector<eosio::name>&& routes, const AMQP::Address& address,
+            std::string&& queue_name)
        : queueName_(std::move(queue_name)), routes_(std::move(routes)) {
-      AMQP::Address amqp_address(address);
-      ilog("Connecting to RabbitMQ address ${a} - Queue: ${q}...", ("a", std::string(amqp_address))("q", queueName_));
+      ilog("Connecting to RabbitMQ address ${a} - Queue: ${q}...", ("a", std::string(address))("q", queueName_));
 
-      init(io_service, std::move(amqp_address));
+      init(io_service, address);
 
       exchangeName_ = DEFAULT_EXCHANGE;
 
       declare_queue();
    }
 
-   rabbitmq(boost::asio::io_service& io_service, std::vector<eosio::name> routes, std::string address, std::string exchange_name, std::string exchange_type)
+   rabbitmq(boost::asio::io_service& io_service, std::vector<eosio::name>&& routes, const AMQP::Address& address,
+            std::string&& exchange_name, const std::string& exchange_type)
```

Comment:
Why change `std::vector` & `std::string` to rvalue `&&` types? By value and move is more appropriate for most "sink" values.
See "Item 41: Consider pass by value for copyable parameters that are cheap to move and always copied" of Effective Modern C++.